include/public/marco/Codegen/Lowering/LoweringContext.h

```diff
@@ -32,10 +34,14 @@ namespace marco::codegen::lowering
 
       mlir::OpBuilder& getBuilder();
 
+      mlir::DiagnosticEngine& getDiagEngine();
```

Comment:
There should be no need for accessing the diagnostic engine at all. Errors can be emitted using the MLIR context.